sdk/core/azure-core/src/http/retry_policy.cpp

```diff
@@ -0,0 +1,170 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// SPDX-License-Identifier: MIT
+
+#include <http/policy.hpp>
+
+#include <algorithm>
+#include <cstdlib>
+#include <limits>
+#include <thread>
+
+using namespace Azure::Core::Http;
+
+namespace {
+typedef decltype(RetryOptions::RetryDelay) Delay;
+typedef decltype(RetryOptions::MaxRetries) RetryNumber;
+
+bool GetResponseHeaderBasedDelay(Response const& response, Delay& retryAfter)
+{
+  // Try to find retry-after headers. There are several of them possible.
+  decltype(response.GetHeaders()) responseHeaders = response.GetHeaders();
```

Comment:
```suggestion
  const auto& responseHeaders = response.GetHeaders();
```

Reply:
I remember now, const references to temporary are fine! I was worried that there might be a problem if return type gets changed from reference to value copy. Now I know it is safe (at least, while it is const)

Reply:
I'll make the change.

Reply:
Fixed.